### PR TITLE
RFC: moving client-side logic into the API

### PR DESCRIPTION
**Pull request by sunnygleason**
Problem:

* too many client-side RPC's
* tricky to navigate app logic
* tricky to manage refresh rates

Solution:

* consolidate work in the API server
* reduce client-side code
* place refresh rates in control of API server

### CODE REVIEW

**Comment by sunnygleason** [comment]
@mvines this websocket refactoring *should* be 1:1 compatible with existing code; I figured I'd land this before plumbing in the validator list view (which shouldn't take long after this since the data's there)

File: api/api.js [hunk 1]
```diff
@@ -101,6 +135,24 @@ let handleTxnRedis = type => (channel, message) => {
 const txnsClient = getClient();
 txnsClient.on('message', handleTxnRedis('txns-by-prgid'));
 
+const globalInfoPublish = handleRedis('global-info');
+
+async function updateGlobalInfoTimerTask() {
+  let globalInfo = await getGlobalInfo();
```
Comment (mvines) [inline api/api.js]: nit: const

File: api/api.js [hunk 2]
```diff
@@ -101,6 +135,24 @@ let handleTxnRedis = type => (channel, message) => {
...
+  globalInfoPublish('global-info', JSON.stringify(globalInfo));
+}
+
+setInterval(updateGlobalInfoTimerTask, GLOBAL_STATS_BROADCAST_INTERVAL_MS);
+
+const clusterInfoPublish = handleRedis('cluster-info');
+
+async function updateClusterInfoTimerTask() {
+  let clusterInfo = await getClusterInfo();
```
Comment (mvines) [inline api/api.js]: nit: const

File: api/api.js [hunk 3]
```diff
@@ -206,7 +258,18 @@ app.get('/global-stats', (req, res) => {
     `!ent-last-id`,
   ];
 
-  sendMgetKeysZipValuesResult(stat_keys, stat_keys, res);
+  let stat_values = await mgetAsync(stat_keys);
```
Comment (mvines) [inline api/api.js]: nit: const

File: api/api.js [hunk 4]
```diff
@@ -206,7 +258,18 @@ app.get('/global-stats', (req, res) => {
...
+
+  return _.zipObject(stat_keys, stat_values);
+}
+
+async function sendGlobalInfoResponse(res) {
+  let globalInfo = await getGlobalInfo();
```
Comment (mvines) [inline api/api.js]: nit: const

**Comment by sunnygleason** [comment]
@mvines thank you!

**Pull request merged**